src/MongoDB.Bson/IO/ByteBufferStream.cs

```diff
@@ -122,6 +123,19 @@ public override long Position
         }
 
         // public methods
+        /// <inheritdoc/>
+        public void EfficientCopyTo(Stream destination)
+        {
+            int remainingCount;
+            while ((remainingCount = (int)(Length - Position)) > 0)
+            {
+                var segment = _buffer.AccessBackingBytes((int)Position);
+                var count = Math.Min(segment.Count, remainingCount);
+                destination.Write(segment.Array, segment.Offset, count);
+                Position += count;
+            }
+        }
```

Comment:
A *slightly* more correct implementation might be:

    long remainingCount;
    while ((remainingCount = Length - Position) > 0)
    {
        var segment = _buffer.AccessBackingBytes((int)Position);
        var count = (int)Math.Min(segment.Count, remainingCount);
        destination.Write(segment.Array, segment.Offset, count);
        Position += count;
    }